lighthouse-core/gather/driver.js

```diff
@@ -743,6 +743,20 @@ class Driver {
   }
 
   /**
+   * Returns the flattened list of all DOM nodes within the document.
+   * @param {boolean=} pierce Whether to pierce through shadow trees and iframes.
+   *     True by default.
+   * @return {!Promise<!Array<!Element>>} The found elements, or [], resolved in a promise
+   */
+  getAllElementsInDocumentAndFrames(pierce = true) {
```

Comment:
I was modeling after the [DT protocol APIs](https://chromedevtools.github.io/devtools-protocol/tot/DOM/#method-getFlattenedDocument), which make `pierce` optional. With the exception that they default to `false`. We want to default to `true`.